controllers/cluster_controller.go

```diff
@@ -684,19 +684,44 @@ func GetPodSpec(cluster *fdbtypes.FoundationDBCluster, processClass fdbtypes.Pro
 	return internal.GetPodSpec(cluster, processClass, idNum)
 }
 
-func (r *FoundationDBClusterReconciler) hasFullReplication(cluster *fdbtypes.FoundationDBCluster) (bool, error) {
-	adminClient, err := r.DatabaseClientProvider.GetAdminClient(cluster, r)
+func hasDesiredFaultTolerance(adminClient AdminClient, cluster *fdbtypes.FoundationDBCluster) (bool, error) {
+	version, err := fdbtypes.ParseFdbVersion(cluster.Spec.Version)
 	if err != nil {
 		return false, err
 	}
-	defer adminClient.Close()
+
+	// TODO (johscheuer): https://github.com/FoundationDB/fdb-kubernetes-operator/issues/743
+	// For FDB versions lower than 6.2.0 we just approve the replacement.
+	// In 6.1 we only have machine information.
+	if !version.IsAtLeast(fdbtypes.FdbVersion{Major: 6, Minor: 2, Patch: 0}) {
+		return true, nil
+	}
 
 	status, err := adminClient.GetStatus()
 	if err != nil {
 		return false, err
 	}
 
-	return status.Cluster.FullReplication, nil
+	if !status.Client.DatabaseStatus.Available {
+		log.V(0).Info("Cluster is not available",
+			"namespace", cluster.Namespace,
+			"cluster", cluster.Name)
+
+		return false, nil
+	}
+
+	expectedFaultTolerance := cluster.DesiredFaultTolerance()
+	log.V(0).Info("Has desired fault tolerance",
```

Comment:
Instead of having the log message make an assertion like "Has desired fault tolerance", could we have it indicate that we're checking the fault tolerance, and include a value in the message that says whether the fault tolerance check is passing? I worry that the current form could give a misleading impression that we have the desired fault tolerance when we really don't.

Reply:
That's a good point, I just used the method name. I'll change that.